loopchain/consensus/syncer.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import time
+from typing import TYPE_CHECKING, Dict
+
+from lft.consensus.events import ReceiveDataEvent, ReceiveVoteEvent
+
+from loopchain import configure as conf
+from loopchain.blockchain.blocks.v1_0 import Block
+from loopchain.blockchain.votes.v1_0 import BlockVote
+from loopchain.channel.channel_property import ChannelProperty
+from loopchain.protos import loopchain_pb2, loopchain_pb2_grpc
+from loopchain.tools.grpc_helper import GRPCHelper
+
+if TYPE_CHECKING:
+    from loopchain.peer.block_manager import BlockManager
+    from lft.event import EventSystem
+
+
+class Syncer:
+    def __init__(self,
+                 block_manager: 'BlockManager',
+                 event_system: 'EventSystem',
+                 last_block_height: int = 0):
+        self._block_manager = block_manager
+        self._event_system: 'EventSystem' = event_system
+        self._last_block_height = last_block_height
+        self._data_info_other_nodes: Dict[int, list] = dict()
+        self._vote_info_other_nodes: Dict[int, list] = dict()
+        self._max_height_in_nodes = last_block_height
+
+        self._request_history_list = dict()
+        self._target_idx = 0
+        self._stub_list = list()
+        self._target_list = list()
+        self.management_stub()
+
+    def management_stub(self):
+        self._target_list = self._block_manager.get_target_list()
+        for target in self._target_list:
+            channel = GRPCHelper().create_client_channel(target)
+            self._stub_list.append(loopchain_pb2_grpc.PeerServiceStub(channel))
+
+    @property
+    def last_block_height(self):
+        return self._last_block_height
+
+    @last_block_height.setter
+    def last_block_height(self, height: int):
+        self._last_block_height = height
+
+    async def sync_start(self):
+        while True:
+            await self._request_block()
+            await self._raise_event()
+
+    async def _request_block(self):
+        def _request(idx: int, height: int):
+            peer_stub = self._stub_list[idx]
+
+            _ = peer_stub.BlockRequest(loopchain_pb2.PeerHeight(
+                peer=ChannelProperty().peer_target,
+                channel=ChannelProperty().name,
+                height=height
+            ), conf.GRPC_TIMEOUT)
+
+        max_loop = self._max_height_in_nodes-3
```

Comment:
What does this code mean?
I can't understand the meaning of `max_loop` and subtraction 3?